### ISSUE TITLE
Bogus analyzer fails with a NPE in the _analyzer API

### ISSUE THREAD

**Issue by costin** [elastic/elasticsearch#17097]
**Elasticsearch version**: 2.2

**JVM version**: 1.8u66

**OS version**: Win 8.1 
 
**Description of the problem including expected versus actual behavior**:

When specifying a bogus analyzer (maybe a typo), ES throws an internal NPE in the logs and the request blocks. Instead the exception should be thrown to the user.

**Steps to reproduce**:
 1. 
```
curl -XPOST http://localhost:9200/_analyze -d '{
  "analyzer" : "bogus",
  "text" : "this is a test"
}'
```
**Provide logs (if relevant)**:
```
[2016-03-14 18:17:44,393][ERROR][transport                ] [Caleb Alexander] failed to handle exception for action [indices:admin/analyze[s]], handler [org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$1@a566c0]
java.lang.NullPointerException
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.perform(TransportSingleShardAction.java:195)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction.access$700(TransportSingleShardAction.java:115)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$1.handleException(TransportSingleShardAction.java:174)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:821)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:799)
	at org.elasticsearch.transport.TransportService$4.onFailure(TransportService.java:361)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:39)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by dadoonet** [comment]
@costin Is it the same as #15148 closed by #15447 in 2.3.0?

**Comment by jasontedor** [comment]
Duplicates #15148, closed by #15447

**Comment by costin** [comment]
looks about right - thanks for jumping on this guys. I just assumed 2.2 is recent enough to not check for duplicates :) Lazy, I know.
